src/js/angular/resources/chart-models/chart-data.js

```diff
@@ -0,0 +1,26 @@
+export class ChartData {
+    constructor(translateService, chartOptions) {
+        this.translateService = translateService;
+        this.range = 150;
+        this.chartOptions = chartOptions;
+        this.chartSetup();
+        this.data = this.createDataHolder();
+    }
+    chartSetup() {}
+    createDataHolder() {
+        throw new Error('Must implement data holder creation');
+    }
+    addData(timestamp, data) {
+        this.removeOldData();
+        this.addNewData(timestamp, data);
+        this.updateRange();
```

Comment:
Maybe pass down all necessary stuff to these template methods in order to make the code less confusing. Right now it's not clear, why in some methods you get things from the class scope and others you get as parameters.

Reply:
Refactored it